src/pages/index.js

```diff
@@ -33,60 +47,165 @@ const cardSection = new Section({
-  jobInput.value = description;
+/* ------------------------------------------------------ */
+/*                    define functions                    */
+/* ------------------------------------------------------ */
+
+function renderProfileDetails(modal){
+  const { name, about } = userInfo.getUserInfo();
+  modal.setInputValues({name, about})
 };
 
 function handleCardAdd(data){
-  const { title, url } = data
-  const cardElement = createCard({title, url})
-  cardSection.addItem(cardElement)
-  cardModal.close()
+  this.renderLoading(true)
+  const { name, link } = data
+  api.addCard(name, link).then((data)=>{
+    cardSection.addItem(createCard(data));
+    formValidators['cardForm'].disableButton();
+    formValidators['cardForm'].resetInputs();
+    cardModal.close()
+  }).catch(err => {
+    console.error(err)
+  }).finally(()=>{
+    this.renderLoading(false)
+  });
 };
 
 function handleImageClick(image){
   const data = {
-    name: image.title,
-    link: image.url
+    name: image.name,
+    link: image.link
   }
   popupWithImage.open(data)
 };
 
+function handleDeleteClick(card){
+  deleteConfirmModal.open()
+  
+  deleteConfirmModal.setSubmitHandler(()=>{
+    deleteConfirmModal.renderLoading(true)
+    api.deleteCard(card.getCardId()).then(()=>{
+      card.delete()
+      deleteConfirmModal.close()
+    }).catch(err => {
+      console.error(err)
+    }).finally(()=>{
+      deleteConfirmModal.renderLoading(false)
+    })
+  })
+}
+
 function createCard(cardData){
-  const card = new Card(cardData, '.card-template', handleImageClick);
+  const card = new Card(cardData, '.card-template', handleImageClick, handleDeleteClick, handleLikeClick, handleDislikeClick);
   return card.generateCard()
 };
 
 function handleProfileFormSubmit(data){
-  userInfo.setUserInfo(data);
+  this.renderLoading(true)
+  const {name, about} = data;
+  api.updateProfile(name, about).then(data => {
+    return data
+  }).then((info) => {
+    userInfo.setUserInfo(info)
+    formValidators['profileForm'].disableButton();
+    formValidators['profileForm'].resetInputs();
+    profileModal.close()
+  }).catch(err => {
+    console.error(err)
+  }).finally(()=>{
+    this.renderLoading(false)
+  })
 }
 
-// Instantiate Classes
-cardAddValidator.enableValidation()
-profileFormValidator.enableValidation()
-cardSection.renderItems();
+function handleLikeClick(id){
+  api.likeCard(id)
+  .then(()=>{
+    this.like()
+  })
+  .catch(err => {
+    console.error(err)
+  })
+}
+
+function handleDislikeClick(id){
+  api.dislikeCard(id)
+  .then(()=> {
+    this.dislike()
+  })
+  .catch(err => {
+    console.error(err)
+  })
+}
+
+function handleAvatarUpdate(info){
+  this.renderLoading(true)
+  api.updateAvatar(info.avatar).then((info)=>{
+    userInfo.setUserInfo(info)
+    formValidators['updateAvatar'].disableButton();
+    formValidators['updateAvatar'].resetInputs();
+    changeAvatarModal.close()
+  }).catch(err => {
+    console.error(err)
```

Comment:
Good job that you `catch` possible errors at the end of server requests. You can simplify it with `.catch(console.error);` because `console.error` is meant to log errors